superset/databases/commands/create.py

```diff
@@ -54,10 +50,9 @@ def run(self) -> Model:
             try:
                 TestConnectionDatabaseCommand(self._actor, self._properties).run()
             except Exception:
-                db.session.rollback()
-                stats_logger.incr(
-                    f"db_connection_failed.{database.db_engine_spec.__name__}"
-                )
+                with event_logger.log_context(action=f"db_connection_failed.{database.db_engine_spec.__name__}"):
+                    db.session.rollback()
```

Comment:
This worries me a bit — if there's an error in the context manager we won't rollback the transaction.

I think it would be better to just log, instead of using a context manager:

```python
try:
    event_logger.log(action=f"db_connection_failed.{database.db_engine_spec.__name__}")
finally:
    db.session.rollback()
```

To ensure that `rollback` is always called.

Reply:
Maybe `log` should be pessimistic / guaranteed to never raise(?) so that we don't have to use caution every time we call it? That would mean wrapping the content of `log` in a `try` ?

Reply:
@betodealmeida  and I were talking about this earlier, and most statements before the yield are pretty safe, but to be sure, I moved what I could below the yield statement. I don't think there's much risk of this failing now:
```
 start_time = time.time()
 payload_override = {}
 ```

Reply:
lmk if you think this works as a solution @betodealmeido @mistercrunch